### DISCUSSION TITLE
how to set svg image in codebehind

### DISCUSSION THREAD

**Discussion by SID9-HoneyBee** [ElinamLLC/SharpVectors#194] [Q&A]
` <InkCanvas x:Name="inkCanvas" EditingMode="None" Width="Auto" Height="Auto">
            <InkCanvas.Background>
                <ImageBrush x:Name="inkCanvasImage"
                            ImageSource="{svgc:SvgImage Source=1.svg}"
                            Stretch="Uniform">
                </ImageBrush>
              
            </InkCanvas.Background>
            <InkCanvas.RenderTransform>
                <TransformGroup>
                    <ScaleTransform x:Name="inkCanvasScale"/>
                    
                    <TranslateTransform x:Name="rectTrans"/>
                </TransformGroup>
                
            </InkCanvas.RenderTransform>
        </InkCanvas>`

I want to set an svg image in an ImageBrush called "inkCanvasImage" .

I have several svg images.

according to the user's keyboard input event
In order to change the image in order
I want to change the ImageSource in codebehind.

Can you tell me how?

**Comment by paulushub** [comment]
I posted a [WpfApp4.zip](https://github.com/ElinamLLC/SharpVectors/files/6557809/WpfApp4.zip) sample in the discussion forum illustrating one way to combine images.

Please take a look at this and see if this works for you.  [SvgViewbox ignores the whitespace around the image](https://github.com/ElinamLLC/SharpVectors/discussions/191)

**Reply by SID9-HoneyBee** [reply]
@paulushub 

Thank you.
Thanks for the advice.

I am displaying multiple SVGs in a list.

It draws very slowly each time you turn the page.

Is preload better?

And there is a very important issue.

When I open the SVG file in Microsoft Edge, all the characters are displayed normally.

However, when I open it in WPF using the library, the characters overlap.

The first screenshot below is from MS Edge and the second screenshot is from the system.

What am I missing?

**Comment by paulushub** [comment]
> It draws very slowly each time you turn the page.
> Is preload better?

Many different options, you might have to experiment to see the best for your use case.
For multiple images, you may also consider using multiple threads, see the **WpfTestThreadSafety** for demo.
And, you can keep your pages alive to prevent create/recreate cases; etc

> When I open the SVG file in Microsoft Edge, all the characters are displayed normally.
> However, when I open it in WPF using the library, the characters overlap.

That could be a **bug**. If you can make public any of your files that will demonstrate this issue then
post this as a **separate issue** not part of this discussion.

**NOTE**: For security, please do not post images with locations of files on your local machine (as in the browser). 
You can cut that part off or blur it.

**Reply by SID9-HoneyBee** [reply]
thank you.

I will post the svg file I used for testing as a separate ticket.

And I removed the image from the comment above because you confirmed it. Thank you.
